autoscale-workload-rabbit/src/main/java/com/github/autoscaler/workload/rabbit/RabbitWorkloadAnalyserFactory.java

```diff
@@ -84,20 +86,38 @@ public WorkloadAnalyser getAnalyser(final String scalingTarget, final String sca
     public HealthResult healthCheck()
     {
         try {
-            if (atLeastOneNodeRunning()) {
-                return HealthResult.RESULT_HEALTHY;
-            } else {
+            if (!atLeastOneNodeRunning()) {
                 final String message = "RabbitMQ management API reports 0 nodes are running: " + nodeStatusEndpoint;
                 LOG.warn(message);
                 return new HealthResult(HealthStatus.UNHEALTHY, message);
+            } else if (config.getIsPayloadOffloadingEnabled() && !offloadingDirectoryExists()) {
+                final String message = "Payload offloading directory is not configured or does not exist.";
+                LOG.warn(message);
+                return new HealthResult(HealthStatus.UNHEALTHY, message);
             }
+            return HealthResult.RESULT_HEALTHY;
         } catch (final IOException | ScalerException e) {
             final String message = "Failed to contact RabbitMQ management API: " + nodeStatusEndpoint;
             LOG.warn(message, e);
             return new HealthResult(HealthStatus.UNHEALTHY, message);
         }
     }
 
+    private boolean offloadingDirectoryExists() throws ScalerException
```

Comment:
I don't think this throws an exception:

```suggestion
    private boolean offloadingDirectoryExists()
```